server/src/main/java/org/opensearch/script/ScoreScriptUtils.java

```diff
@@ -69,6 +73,90 @@ public static double sigmoid(double value, double k, double a) {
         return Math.pow(value, a) / (Math.pow(k, a) + Math.pow(value, a));
     }
 
+    /**
+     * Retrieves the term frequency within a field for a specific term.
+     *
+     * @opensearch.internal
+     */
+    public static final class TermFreq {
+        private final ScoreScript scoreScript;
+
+        public TermFreq(ScoreScript scoreScript) {
+            this.scoreScript = scoreScript;
+        }
+
+        public int termFreq(String field, String term) {
+            try {
+                return (int) scoreScript.getTermFrequency(TERM_FREQ, field, term);
+            } catch (Exception e) {
+                throw ExceptionsHelper.convertToOpenSearchException(e);
+            }
+        }
+    }
+
+    /**
+     * Calculates the term frequency-inverse document frequency (tf-idf) for a specific term within a field.
+     *
+     * @opensearch.internal
+     */
+    public static final class TF {
+        private final ScoreScript scoreScript;
+
+        public TF(ScoreScript scoreScript) {
+            this.scoreScript = scoreScript;
+        }
+
+        public float tf(String field, String term) {
+            try {
+                return (float) scoreScript.getTermFrequency(TF, field, term);
+            } catch (Exception e) {
+                throw ExceptionsHelper.convertToOpenSearchException(e);
+            }
+        }
+    }
+
+    /**
+     * Retrieves the total term frequency within a field for a specific term.
+     *
+     * @opensearch.internal
+     */
+    public static final class TotalTermFreq {
+        private final ScoreScript scoreScript;
+
+        public TotalTermFreq(ScoreScript scoreScript) {
+            this.scoreScript = scoreScript;
+        }
+
+        public long totalTermFreq(String field, String term) {
+            try {
+                return (long) scoreScript.getTermFrequency(TOTAL_TERM_FREQ, field, term);
+            } catch (Exception e) {
+                throw ExceptionsHelper.convertToOpenSearchException(e);
+            }
+        }
+    }
+
+    /**
+     * Retrieves the sum of total term frequencies within a field.
+     *
+     * @opensearch.internal
+     */
+    public static final class SumTotalTermFreq {
+        private final ScoreScript scoreScript;
+
+        public SumTotalTermFreq(ScoreScript scoreScript) {
+            this.scoreScript = scoreScript;
+        }
+
+        public long sumTotalTermFreq(String field) {
+            try {
+                return (long) scoreScript.getTermFrequency(SUM_TOTAL_TERM_FREQ, field, null);
+            } catch (Exception e) {
+                throw ExceptionsHelper.convertToOpenSearchException(e);
+            }
+        }
+    }
+
```

Comment:
Wondering if these can belong to common static class and mapped as such in painless spi

Reply:
Painless doesn't support binding multiple methods.

```
=== Standard error of node `node{::runTask-0}` ===
»   ↓ last 40 non error or warning messages from /Volumes/workplace/OpenSearch/build/testclusters/runTask-0/logs/opensearch.stderr.log ↓
» uncaught exception in thread [main]
»  java.lang.IllegalArgumentException: error loading allowlist(s) [org.opensearch.score.txt]:[26]
»  Likely root cause: java.lang.IllegalArgumentException: class binding [org.opensearch.script.ScoreScriptUtils.TermFreq] cannot have multiple methods
»       at org.opensearch.painless.lookup.PainlessLookupBuilder.addPainlessClassBinding(PainlessLookupBuilder.java:1534)
»       at org.opensearch.painless.lookup.PainlessLookupBuilder.addPainlessClassBinding(PainlessLookupBuilder.java:1420)
»       at org.opensearch.painless.lookup.PainlessLookupBuilder.buildFromAllowlists(PainlessLookupBuilder.java:195)
»       at org.opensearch.painless.PainlessScriptEngine.<init>(PainlessScriptEngine.java:113)
»       at org.opensearch.painless.PainlessModulePlugin.getScriptEngine(PainlessModulePlugin.java:127)
»       at org.opensearch.script.ScriptModule.<init>(ScriptModule.java:100)
»       at org.opensearch.node.Node.<init>(Node.java:543)
»       at org.opensearch.node.Node.<init>(Node.java:388)
»       at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242)
»       at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242)
»       at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404)
»       at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180)
»       at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171)
»       at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104)
»       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
»       at org.opensearch.cli.Command.main(Command.java:101)
»       at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137)
»       at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103)
»  For complete error details, refer to the log at /Volumes/workplace/OpenSearch/build/testclusters/runTask-0/logs/runTask.log
```